### ISSUE TITLE
:bulb: Make it possible to put remarks (entities and relationships) :memo: 

### ISSUE THREAD

**Issue by adriens** [neo4j/neo4j#13198]
# :grey_question: About

We currently run an ever bigger and complex Neo4J database as we use it to put[ our whole information system cartography in it](https://dev.to/optnc/our-speech-about-it-holism-at-nodes22-1bpl).

When comes the time to document our relational (eg. PostrgeSQL) database, we use to (through [`Liquibase`](https://docs.liquibase.com/change-types/set-table-remarks.html)) : 

- :heavy_check_mark: Put [remarks/comments ](https://www.postgresql.org/docs/current/sql-comment.html)on all database objects (table, columns,...)
- :heavy_check_mark: generate a graph with [schemacrawler](https://www.schemacrawler.com/) that uses previous metadatas

The benfits are that we could totally automate our documentation process from code, within CI from end to end.

:point_right: We would like to achieve the same automated documentation workflow on Neo4J... but currently it looks like we are stuck in the following situation : 

- ❌ Put remarks on all database objects (entities & relationships
- :heavy_check_mark: Generate database diagram with [`apoc.meta.graph`](https://neo4j.com/labs/apoc/4.3/overview/apoc.meta/apoc.meta.graph/)

# :pray: Feature request

Make it possible to put remarks (from within `cypher` so we can embed it within [`liquibase/liquibase-neo4j`](https://github.com/liquibase/liquibase-neo4j) ) on any Neo4J object: 

- [ ] entities
- [ ] relatrionsships
- [ ] constraints
- [ ] indexes
- [ ] UDF
- [x] Any object... including graphes themselves (already implement I guess :thought_balloon: )

# :bookmark: Resources

- [🌌 Digital identities journey w. Neo4J ](https://dev.to/optnc/digital-identities-journey-w-neo4j-5gep)
- [🕵️ Active Directory Groups audit w. Neo4J](https://dev.to/optnc/active-directory-groups-audit-w-neo4j-483o)
- [🗣️ Our speech about "IT holism" at #nodes22 💡](https://dev.to/optnc/our-speech-about-it-holism-at-nodes22-1bpl)
- [COMMENT on psql](https://www.postgresql.org/docs/current/sql-comment.html)
- [Liquibase `setColumnRemarks`](https://docs.liquibase.com/change-types/set-column-remarks.html)
- [Liquibase `setTableRemarks`](https://docs.liquibase.com/change-types/set-table-remarks.html)

# :weight_lifting: Current alternative

We currelty achieve a manuel documentation process :sweat: .

**Comment by JonasHolgersson** [comment]
Thanks for the report, I have forwarded your request to the product team.

Regards,
Jonas

**Comment by sherfert** [comment]
@adriens Thank you for the feature request.

I am trying to understand if the following could work as a workaround/alternative:

Instead of comments on schema objects themselves, you create nodes representing those schema objects and put the remarks into properties of these nodes. Let's say you have a Label `Foo` and want to add some documentation to it.
You create a node `(:Label {labelName: "Foo", documentation: "Some text"})`. 
Would that work as a workaround/alternative for you?

**Comment by adriens** [comment]
> Instead of comments on schema objects themselves, you create nodes representing those schema objects and put the remarks into properties of these nodes. 

Yes, I first thought about it but the drawback is that is has an impact on the graph topology itself and also may complexify it.
With this approach, I may create : 

1. Nodes for each label
2. Nodes for each Node
3. Node for each relationship label
4. Node for each relationship

... which could lead in fact as a whole apart and dedicated graph, it's an approach, but I would have rather been able to generate it from initial graph vs. manually maintaining a new one.... which leads into inaccurate datas.

**Comment by adriens** [comment]
[post](https://www.linkedin.com/feed/update/urn:li:activity:7085198234331033601?commentUrn=urn%3Ali%3Acomment%3A%28activity%3A7085198234331033601%2C7085522031538040832%29&dashCommentUrn=urn%3Ali%3Afsd_comment%3A%287085522031538040832%2Curn%3Ali%3Aactivity%3A7085198234331033601%29)
![image](https://github.com/neo4j/neo4j/assets/5235127/e090e17f-b793-40c4-8e4e-bcc07b3d7c62)
